code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1298 @@
+		else
+			msg += "It's holding [bicon(I)] \a [I] in its [get_index_limb_name(is_holding_item(I))].\n"
+
+	//gloves
+	if(clothing[SLOT_MANNEQUIN_GLOVES])
+		var/obj/item/gloves = clothing[SLOT_MANNEQUIN_GLOVES]
+		if(gloves.blood_DNA && gloves.blood_DNA.len)
+			msg += "<span class='warning'>It has [bicon(gloves)] [gloves.gender==PLURAL?"some":"a"] blood-stained [gloves.name] on its hands![gloves.description_accessories()]</span>\n"
+		else
+			msg += "It has [bicon(gloves)] \a [gloves] on its hands.[gloves.description_accessories()]\n"
+
+	//belt
+	if(clothing[SLOT_MANNEQUIN_BELT])
+		var/obj/item/belt = clothing[SLOT_MANNEQUIN_BELT]
+		if(belt.blood_DNA && belt.blood_DNA.len)
+			msg += "<span class='warning'>It has [bicon(belt)] [belt.gender==PLURAL?"some":"a"] blood-stained [belt.name] about its waist![belt.description_accessories()]</span>\n"
+		else
+			msg += "It has [bicon(belt)] \a [belt] about its waist.[belt.description_accessories()]\n"
+
+	//shoes
+	if(clothing[SLOT_MANNEQUIN_FEET])
+		var/obj/item/shoes = clothing[SLOT_MANNEQUIN_FEET]
+		if(shoes.blood_DNA && shoes.blood_DNA.len)
+			msg += "<span class='warning'>It's wearing [bicon(shoes)] [shoes.gender==PLURAL?"some":"a"] blood-stained [shoes.name] on its feet![shoes.description_accessories()]</span>\n"
+		else
+			msg += "It's wearing [bicon(shoes)] \a [shoes] on its feet.[shoes.description_accessories()]\n"
+
+	//mask
+	if(clothing[SLOT_MANNEQUIN_MASK])
+		var/obj/item/wear_mask = clothing[SLOT_MANNEQUIN_MASK]
+		if(wear_mask.blood_DNA && wear_mask.blood_DNA.len)
+			msg += "<span class='warning'>It has [bicon(wear_mask)] [wear_mask.gender==PLURAL?"some":"a"] blood-stained [wear_mask.name] on its face![wear_mask.description_accessories()]</span>\n"
+		else
+			msg += "It has [bicon(wear_mask)] \a [wear_mask] on its face.[wear_mask.description_accessories()]\n"
+
+	//eyes
+	if(clothing[SLOT_MANNEQUIN_EYES])
+		var/obj/item/glasses = clothing[SLOT_MANNEQUIN_EYES]
+		if(glasses.blood_DNA && glasses.blood_DNA.len)
+			msg += "<span class='warning'>It has [bicon(glasses)] [glasses.gender==PLURAL?"some":"a"] blood-stained [glasses] covering its eyes![glasses.description_accessories()]</span>\n"
+		else
+			msg += "It has [bicon(glasses)] \a [glasses] covering its eyes.[glasses.description_accessories()]\n"
+
+	//ears
+	if(clothing[SLOT_MANNEQUIN_EARS])
+		var/obj/item/ears = clothing[SLOT_MANNEQUIN_EARS]
+		msg += "It has [bicon(ears)] \a [ears] on its ears.[ears.description_accessories()]\n"
+
+	//ID
+	if(clothing[SLOT_MANNEQUIN_ID])
+		var/obj/item/wear_id = clothing[SLOT_MANNEQUIN_ID]
+		msg += "It's wearing [bicon(wear_id)] \a [wear_id].\n"
+
+	to_chat(user, msg)
+
+
+/obj/structure/mannequin/ex_act(severity)
+	switch(severity)
+		if (1)
+			qdel(src)
+		if (2)
+			if (prob(50))
+				health -= 30
+			else
+				health -= 20
+			healthcheck()
+		if (3)
+			if (prob(50))
+				health -= 10
+				healthcheck()
```

Comment:
Rather than requiring a healthcheck proc separately, it would be saner to have a proc for changing health that did the checks for you.